marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java

```diff
@@ -120,24 +120,31 @@ public void accept(InputStream input) {
 				throw new MarkLogicInternalException(
 						"unexpected state for " + getEndpointPath() + " during loop: " + getPhase().name());
 			}
+
+			logger.trace("input endpoint running endpoint={} count={} state={}", getEndpointPath(), getCallCount(),
+					getEndpointState());
+			processInput(queue);
+		}
+
+		@Override
+		public void awaitCompletion() {
+			logger.trace("input endpoint running endpoint={} work={}", getEndpointPath(), getWorkUnit());
+			if(queue != null && !queue.isEmpty())
+				processInput(queue);
+		}
+
+		private void processInput(LinkedBlockingQueue<InputStream> queue) {
```

Comment:
Is there a rationale for passing the queue as a parameter instead of accessing it locally?

By the same reasoning, is there a rationale for logging in the caller instead of locally?

If not, maybe it would be simpler and easier to maintain to access the queue and to log locally within processInput().

Reply:
Thanks Erik.!
Added the changes.